libvast/vast/table_slice.hpp

```diff
@@ -164,6 +164,17 @@ class table_slice final {
   /// @pre `row < rows() && column < columns()`
   data_view at(size_type row, size_type column) const;
 
+#if VAST_HAVE_ARROW
+
+  /// Converts a table slice to an Apache Arrow Record Batch.
+  /// @returns The pointer to the Record Batch on success, or `nullptr`.
+  /// @param x The table slice to convert.
+  /// @note Returns `nullptr` if `x.encoding() != table_slice::encoding::arrow`.
+  friend std::shared_ptr<arrow::RecordBatch>
+  as_record_batch(const table_slice& x);
```

Comment:
I think keeping interface lean is more important than optimizing for ADL overload size here. Could this be moved back to the writer?

Reply:
It can't be, it must access internals.

Reply:
We decided to introduce `as_arrow_table_slice` and `as_msgpack_table_slice` functions instead.